torch/lib/c10d/Utils.hpp

```diff
@@ -392,6 +392,15 @@ std::vector<T*> getDataPointers(const std::vector<at::Tensor>& tensors) {
   return ptrs;
 }
 
+// Helper to check whether t1 and t2 share the same storage
+inline bool shareSameStorage(const at::Tensor& t1, const at::Tensor& t2) {
```

Comment:
Storage has a very specific meaning in pytorch as Storage is a type of elementary object so I don't think you want to use it here.

Do you actually want to check if the two Tensors can potentially look at the same memory? If so, we already have function to check if Tensors alias each other.

Otherwise, could you explain what this function is checking for exactly? In particular, it is very surprising that it checks for things like numel and stride but not size.

Reply:
@albanD thanks for commenting! 

@mcarilli reminded me that "is_alias_of()" returning true as long as the underlying storage is the same for two tensors (even if they view different regions of the storage). 

In this case, "is_alias_of()" seems not to be the right method used in DDP. In DDP, we allocate a large tensor e.g., called Bucket. we've let grad1 point to one slice of the Bucket called bucket_view_1, and let grad2 point to another slice of the Bucket called bucket_view_2.  

What we want is: shareSameStorage(grad2, bucket_view_1) return False, However, grad2.is_alias_of(bucket_view_1) will return True. 

That is why I'm trying to write this method to make sure shareSameStorage(grad2, bucket_view_1) return False. 

Does it make sense to you?  agree that the naming is not proper, also I'm not confident that what I'm checking is the right way. That is why I'm requesting your review and feedback about how to write the right helper method to check two tensors are the same (or, am I overthinking?  Will t1==t2 work..?)

Reply:
Could you give a full description on where this function should return True and False?